packages/ui/src/views/project/project-members/components/member-list.tsx

```diff
@@ -0,0 +1,129 @@
+              <div className="flex gap-1.5">
+                <Text wrap="nowrap" size={1} truncate className="text-tertiary-background">
+                  {member.email}
+                </Text>
+              </div>
+            </TableCell>
+            {/* ROLE */}
+            <TableCell className="my-6 content-center">
+              <DropdownMenu>
+                <DropdownMenuTrigger className="flex items-center gap-x-1.5">
+                  {upperFirst(member.role)}
+                  <Icon className="chevron-down text-icons-4" name="chevron-fill-down" size={6} />
+                </DropdownMenuTrigger>
+                <DropdownMenuContent className="w-[300px]">
+                  <DropdownMenuItem
+                    className="flex flex-col"
+                    key="owner"
+                    onClick={() => onEdit({ ...member, role: 'space_owner' })}
+                  >
+                    <Text className="inline-block w-full text-left">Owner</Text>
+                    <Text className="mt-1.5 inline-block w-full text-muted-foreground">
+                      Admin-level access to all resources.
+                    </Text>
+                  </DropdownMenuItem>
+                  <DropdownMenuItem
+                    className="flex flex-col"
+                    key="contributor"
+                    onClick={() => onEdit({ ...member, role: 'contributor' })}
+                  >
+                    <Text className="inline-block w-full text-left">Contributor</Text>
+                    <Text className="mt-1.5 inline-block w-full text-muted-foreground">
+                      Can view, comment, and edit resources.
+                    </Text>
+                  </DropdownMenuItem>
+                  <DropdownMenuItem
+                    className="flex flex-col"
+                    key="reader"
+                    onClick={() => onEdit({ ...member, role: 'reader' })}
+                  >
+                    <Text className="inline-block w-full text-left">Reader</Text>
+                    <Text className="mt-1.5 inline-block w-full text-muted-foreground">Can view and comment.</Text>
+                  </DropdownMenuItem>
+                  <DropdownMenuItem
+                    className="flex flex-col"
+                    key="executor"
+                    onClick={() => onEdit({ ...member, role: 'executor' })}
+                  >
+                    <Text className="inline-block w-full text-left">Executor</Text>
+                    <Text className="mt-1.5 inline-block w-full text-muted-foreground">
+                      Can view but cannot make changes or leave comments.
+                    </Text>
+                  </DropdownMenuItem>
+                </DropdownMenuContent>
+              </DropdownMenu>
+            </TableCell>
+            <TableCell className="my-6 content-center">
+              <div className="flex items-center justify-end gap-1.5">{moreActionsDropdown({ member, onDelete })}</div>
```

Comment:
This should be component.

Reply:
Done in https://github.com/harness/canary/pull/613.